source/core/shape.cc

```diff
@@ -1660,6 +1707,217 @@ RBFShape<dim>::initialize_bounding_box()
+  std::iota(std::begin(*temporary_node_numbers),
+            std::end(*temporary_node_numbers),
+            0);
+  std::map<const typename DoFHandler<dim>::cell_iterator,
+           std::tuple<Point<dim>, double, std::shared_ptr<std::vector<size_t>>>>
+    temporary_nodes_portions_map;
+  temporary_nodes_portions_map.clear();
+  for (unsigned int level = 0; level + levels_not_precalculated < maximal_level;
+       level++)
+    {
+      max_number_of_inside_nodes = 1;
+      const auto &cell_iterator  = dof_handler.cell_iterators_on_level(level);
+      for (const auto &cell : cell_iterator)
+        {
+          if (cell->is_artificial_on_level())
+            continue;
+          if (cell->level() > 0)
+            try
+              {
+                const auto cell_parent = cell->parent();
+                const auto parent_iterator =
+                  temporary_nodes_portions_map.find(cell_parent);
+                if (parent_iterator != temporary_nodes_portions_map.end())
+                  {
+                    temporary_node_numbers =
+                      std::get<2>(parent_iterator->second);
+                  }
+                else
+                  // If the parent doesn't contain any node, the child will not
+                  // either
+                  continue;
+              }
+            catch (TriaAccessorExceptions::ExcCellHasNoParent())
+              {}
+
+          std::shared_ptr<std::vector<size_t>> current_cell_nodes =
+            std::make_shared<std::vector<size_t>>();
+          current_cell_nodes->reserve(max_number_of_inside_nodes);
+          for (unsigned int i = 0; i < temporary_node_numbers->size(); i++)
+            {
+              if (cell->point_inside(
+                    rotated_nodes_positions[(*temporary_node_numbers)[i]]))
+                current_cell_nodes->push_back(temporary_node_numbers->at(i));
+            }
+          max_number_of_inside_nodes =
+            std::max(max_number_of_inside_nodes, current_cell_nodes->size());
+          current_cell_nodes->shrink_to_fit();
+          if (current_cell_nodes->size() > 0)
+            temporary_nodes_portions_map[cell] =
+              std::make_tuple(cell->barycenter(),
+                              cell->diameter(),
+                              current_cell_nodes);
+
+          // We currently ignore RBF nodes that are in no cells
+          // A fix for this will be implemented shortly
+        }
+      // We need to remove all cells that are not needed anymore
+      // from the map
+      for (auto it = temporary_nodes_portions_map.cbegin();
+           it != temporary_nodes_portions_map.cend();)
+        {
+          auto               cell       = it->first;
+          const unsigned int cell_level = cell->level();
+          bool cell_still_needed = cell->is_active() || cell_level == level;
+          if (!cell_still_needed)
+            {
+              temporary_nodes_portions_map.erase(it++->first);
```

Comment:
Good question. I searched a bit, but from my understanding we have to iterate before we remove the entry that the iterator refers to, otherwise the iterator becomes invalidated. I changed it a bit so that it's more readable, but I still have to place it++ inside the loop, at the latest when I call erase().
https://stackoverflow.com/questions/23353812/does-stdmaperaseit-maintain-a-valid-iterator-pointing-to-the-next-elemen